### PR TITLE
Initialize bools in the headers in `editor/`

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR is a codestyle enhancement which initializes bools in the headers instead of the constructors. I only did this for the `editor/` folder. I also deleted a few unused bools - future PRs can add them back if they're needed.

### CODE REVIEW

**Review by fire — COMMENTED** [review]
Wanted to mention it was error prone to find all the booleans that were true in other refactors.

File: editor/animation_track_editor_plugins.h [hunk 1]
```diff
@@ -114,7 +114,7 @@ class AnimationTrackEditTypeAudio : public AnimationTrackEdit {
 
 	void _preview_changed(ObjectID p_which);
 
-	bool len_resizing;
+	bool len_resizing = false;
 	bool len_resizing_start;
```
Comment (akien-mga) [inline editor/animation_track_editor_plugins.h:118]: Missed this one.

File: editor/editor_file_dialog.h [hunk 2]
```diff
@@ -145,8 +145,8 @@ class EditorFileDialog : public ConfirmationDialog {
 	bool show_hidden_files;
```
Comment (akien-mga) [inline editor/editor_file_dialog.h:145]: This one too.

Comment (akien-mga) [reply editor/editor_file_dialog.h:145]: (Might be worth doing a second pass grepping for `bool [a-zA-Z0-9_]+;` in headers I guess. Could be a follow up PR if there are many.)

File: editor/scene_tree_editor.h [hunk 3]
```diff
@@ -92,18 +92,18 @@ class SceneTreeEditor : public Control {
 
 	bool can_rename;
```
Comment (akien-mga) [inline editor/scene_tree_editor.h:93]: These too.

File: editor/animation_track_editor_plugins.h (hunk 1 above)
Comment (aaronfranke) [reply editor/animation_track_editor_plugins.h:118]: For now I haven't touched ones that aren't already initialized in the constructor. But I could go and ensure all of the bools that currently aren't initialized are initialized to false.

**Review by akien-mga — APPROVED** [review]

File: editor/animation_track_editor_plugins.h (hunk 1 above)
Comment (akien-mga) [reply editor/animation_track_editor_plugins.h:118]: Let'S do it as a follow up I guess, this one is good as is.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**